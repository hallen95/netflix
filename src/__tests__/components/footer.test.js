import { render } from '@testing-library/react'
import { Footer } from '../../components'

describe('<Footer />', () => {
    it('renders the <Footer /> with populated data', () => {
        const { container, getByText } = render( <Footer>
            <Footer.Title> Question? Contact us</Footer.Title>
            <Footer.Break />
            <Footer.Row>
                <Footer.Column>
                    <Footer.Link href="#">FAQ</Footer.Link>
                    <Footer.Link href="#">Investor Relations</Footer.Link>
                    <Footer.Link href="#">Ways to watch</Footer.Link>
                    <Footer.Link href="#">Corporate Information</Footer.Link>
                    <Footer.Link href="#">Netflix Originals</Footer.Link>
                </Footer.Column>

                <Footer.Column>
                    <Footer.Link href="#">Help center</Footer.Link>
                    <Footer.Link href="#">Jobs</Footer.Link>
                    <Footer.Link href="#">Terms of use</Footer.Link>
                    <Footer.Link href="#">Contact Us</Footer.Link>
                </Footer.Column>

                <Footer.Column>
                    <Footer.Link href="#">Account</Footer.Link>
                    <Footer.Link href="#">Redeem gift cards</Footer.Link>
                    <Footer.Link href="#">Privacy</Footer.Link>
                    <Footer.Link href="#">Speed Test</Footer.Link>
                </Footer.Column>

                <Footer.Column>
                    <Footer.Link href="#">Media center</Footer.Link>
                    <Footer.Link href="#">Buy gift cards</Footer.Link>
                    <Footer.Link href="#">Cookie preferences</Footer.Link>
                    <Footer.Link href="#">Legal notices</Footer.Link>
                </Footer.Column>
            </Footer.Row>
            <Footer.Break />
            <Footer.Text>Netflix Argentina</Footer.Text>
        </Footer>)
        expect(getByText('Question? Contact us')).toBeTruthy()
        expect(getByText('FAQ')).toBeTruthy()
        expect(getByText('Investor Relations')).toBeTruthy()
        expect(getByText('Ways to watch')).toBeTruthy()
        expect(getByText('Corporate Information')).toBeTruthy()
        expect(getByText('Netflix Originals')).toBeTruthy()
        expect(container.firstChild).toMatchSnapshot()
    })
} )