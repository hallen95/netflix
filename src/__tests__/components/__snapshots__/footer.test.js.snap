// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-sc-1kzmoke-0 eVDKzJ"
>
  <p
    class="footer__Title-sc-1kzmoke-4 dZSKrM"
  >
     Question? Contact us
  </p>
  <div
    class="footer__Break-sc-1kzmoke-6 YiCKL"
  />
  <div
    class="footer__Row-sc-1kzmoke-2 gvVAam"
  >
    <div
      class="footer__Column-sc-1kzmoke-1 hqAXEl"
    >
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Ways to watch
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-sc-1kzmoke-1 hqAXEl"
    >
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Help center
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Terms of use
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-sc-1kzmoke-1 hqAXEl"
    >
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Redeem gift cards
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-sc-1kzmoke-1 hqAXEl"
    >
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Media center
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Buy gift cards
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Cookie preferences
      </a>
      <a
        class="footer__Link-sc-1kzmoke-3 gpGATd"
        href="#"
      >
        Legal notices
      </a>
    </div>
  </div>
  <div
    class="footer__Break-sc-1kzmoke-6 YiCKL"
  />
  <p
    class="footer__Text-sc-1kzmoke-5 jZetlx"
  >
    Netflix Argentina
  </p>
</div>
`;
